bundles/ui/org.eclipse.smarthome.ui.classic/src/main/java/org/eclipse/smarthome/ui/classic/internal/render/AbstractWidgetRenderer.java

```diff
@@ -158,4 +164,37 @@ protected String processColor(Widget w, String snippet) {
 
         return snippet;
     }
+
+    protected String getFormat() {
+        return "svg";
+    }
+
+    protected String getState(Widget w) {
+        State state = itemUIRegistry.getState(w);
+        if (state != null) {
+            return escapeURLPath(state.toString());
+        } else {
+            return "NULL";
+        }
+    }
+
+    protected String getStateAsNumber(Widget w) {
+        String itemName = w.getItem();
+        if (itemName != null) {
+            try {
+                Item item = itemUIRegistry.getItem(itemName);
+                return escapeURLPath(item.getStateAs(DecimalType.class).toString());
+            } catch (ItemNotFoundException e) {
+                logger.error("Cannot retrieve item '{}' for widget {}",
+                        new Object[] { itemName, w.eClass().getInstanceTypeName() });
+            }
+        }
+        return "NULL";
+    }
+
+    protected String getIcon(Widget w) {
```

Comment:
Same question here: What does it actually return? a category, an Icon (id)? I think the concrete item is at the end calculated out of the category, the pattern and the state. So "getIcon" might be a bit misleading.